### ISSUE TITLE
Sphinx inherited methods

### ISSUE THREAD

**Issue by kbarnhart** [landlab/landlab#832]
As discussed in PR #822 it would be good to have the documentation for classes that inherit from base classes get their documentation (e.g. LossyFlowAccumulator's run_one_step). 

[This stackoverflow might help...](https://stackoverflow.com/questions/40508492/python-sphinx-inherit-method-documentation-from-superclass)

**Comment by SiccarPoint** [comment]
It's been an age since I built the files to produce the grid docs, and scanning I think @kbarnhart has done an excellent job revising them since. However, I don't think this is a problem with the doc builder functionality.

When the core team restructured the grids to inherit from the graphs, it looks like the docstrings in the graphs never received LLCATS at the end, which is the system Landlab uses to flag what kind of method each is when it creates the tables e.g. here: https://landlab.readthedocs.io/en/latest/reference/grid/raster.html.

This was all done because a giant list of all grid methods is really unhelpful, but it does put the onus on the developer to flag what each method or property does in its docstring, using the LLCATS system. For example, the `axis_name` property in `base.py` looks like this, and so gets listed under "Information about the grid as a whole" in the webpages:
```
    def axis_name(self):
        """Get the name of each coordinate axis.

        Returns
        -------
        tuple of str
            The names of each axis.

        Examples
        --------
        >>> from landlab import RasterModelGrid
        >>> grid = RasterModelGrid((4, 5))
        >>> grid.axis_name
        ('x', 'y')
        >>> grid.axis_name = ('lon', 'lat')
        >>> grid.axis_name
        ('lon', 'lat')

        LLCATS: GINF
        """
        ...
```
GINF here is specifying the section.

**TLDR:** I believe if LLCATS are placed in the docstrings on the key inherited methods in the graph methods, they will be automatically picked up and documented.

For reference, the LLCATS are
```
LLCATS = (
    "GINF",
    "NINF",
    "LINF",
    "CINF",
    "PINF",
    "FINF",
    "CNINF",
    "GRAD",
    "MAP",
    "BC",
    "SUBSET",
    "SURF",
)
```
...as in `docs/build_grid_func_docs.py`. These map one-to-one onto the categories on the grid webpages.

(I note that the problem in the components is largely fixed, as they all (most??) declare `:inherited-members:` now. e.g., `lossy_flow_accumulator` now appears to have `run_one_step`. Someone must have fixed this in 2018. Maybe me? Who knows. Was probably a search-and-replace job. See here: https://www.sphinx-doc.org/en/master/usage/extensions/autodoc.html. But this is not how the grid pages are built...)
